GitCommands/Git/Commands/GitCommandHelpers.cs

```diff
@@ -109,9 +109,7 @@ public static ArgumentString GetRefsCmd(bool tags, bool branches, bool noLocks,
             {
                 { sortBy != GitRefsSortBy.Default, GetSortCriteria(tags, sortBy, sortOrder), string.Empty },
                 format,
-                { branches, "refs/heads/", string.Empty },
-                { branches && tags, "refs/remotes/", string.Empty },
-                { tags, "refs/tags/", string.Empty },
+                { !tags, "refs/remotes/ refs/heads/", string.Empty },
```

Comment:
This is incorrect, since this would never be true.

Since we can't filter at `` commands, we can fix it by delegating the filter to the bash:
```diff
                 format,
-                { branches, "refs/heads/", string.Empty },
-                { branches && tags, "refs/remotes/", string.Empty },
-                { tags, "refs/tags/", string.Empty },
+                { !branches, "| sed 'refs\\/heads\\//d", string.Empty },
+                { !(branches && tags), "| sed 'refs\\/remotes\\//d", string.Empty },
+                { !tags, "| sed 'refs\\/tags\\//d", string.Empty },
             };
```

Reply:
Here's the fix for the tests:
```diff
diff --git a/UnitTests/GitCommands.Tests/Git/Commands/GitCommandHelpersTest.cs b/UnitTests/GitCommands.Tests/Git/Commands/GitCommandHelpersTest.cs
index ec2ab688c..62e35dee9 100644
--- a/UnitTests/GitCommands.Tests/Git/Commands/GitCommandHelpersTest.cs
+++ b/UnitTests/GitCommands.Tests/Git/Commands/GitCommandHelpersTest.cs
@@ -664,13 +664,14 @@ private static IEnumerable<TestCaseData> GetRefsCommandTestData
                         }
 
                         yield return new TestCaseData(/* tags */ true, /* branches */ true, /* noLocks */ false, sortBy, sortOrder,
-                            /* expected */ $@"for-each-ref{sortConditionRef}{sortCondition}{format} refs/heads/ refs/remotes/ refs/tags/");
+                            /* expected */ $@"for-each-ref{sortConditionRef}{sortCondition}{format}");
                         yield return new TestCaseData(/* tags */ true, /* branches */ false, /* noLocks */ false, sortBy, sortOrder,
-                            /* expected */ $@"for-each-ref{sortConditionRef}{sortCondition}{format} refs/tags/");
+                            /* expected */ $@"for-each-ref{sortConditionRef}{sortCondition}{format} | sed 'refs\/heads\//d | sed 'refs\/remotes\//d");
                         yield return new TestCaseData(/* tags */ false, /* branches */ true, /* noLocks */ false, sortBy, sortOrder,
-                            /* expected */ $@"for-each-ref{sortCondition} --format=""%(objectname) %(refname)"" refs/heads/");
+                            /* expected */ $@"for-each-ref{sortCondition} --format=""%(objectname) %(refname)"" | sed 'refs\/remotes\//d | sed 'refs\/tags\//d");
+
                         yield return new TestCaseData(/* tags */ false, /* branches */ true, /* noLocks */ true, sortBy, sortOrder,
-                            /* expected */ $@"--no-optional-locks for-each-ref{sortCondition} --format=""%(objectname) %(refname)"" refs/heads/");
+                            /* expected */ $@"--no-optional-locks for-each-ref{sortCondition} --format=""%(objectname) %(refname)"" | sed 'refs\/remotes\//d | sed 'refs\/tags\//d");
                     }
                 }
             }
```

Reply:
The argument need to be added twice, in some other way. But the matching should be done on a positional base, sed will match in the middle of the refs too

Reply:
I changed the solution to handle the RevGrid usage specially and explicitly, to really show that this is a special usage.
I looked at the usage for when both tags and branches were retrieved, they were all to get remotes - that should be a separate option too, but I do not want to change the IGitModule interface. Added some xml doc.

Note that stash, notes, bisect labels are always shown.

Reply:
> The argument need to be added twice, in some other way. But the matching should be done on a positional base, sed will match in the middle of the refs too

To match the start of line use `^`

https://www.howtogeek.com/666395/how-to-use-the-sed-command-on-linux/

Reply:
> The argument need to be added twice, in some other way. 

Can you please clarify this bit?